### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by adarsh**
Reason for Change
=================
* The documentation needed some grooming.

Changes
=======
* Move the instructions for contributing to the application into `CONTRIBUTING.md`.
* Add a document index to the top of the `README.md`.
* Misc copyediting and reordering.

Minor
-----
* Copy edits for `bin/setup`.

### CODE REVIEW

**Comment by adarsh** [comment]
Reviewers: be sure to use the rich diff option in GitHub:

image

**Comment by adarsh** [comment]
[Socialize a Bug Triage process](https://trello.com/c/JFdiqjHR/297-socialize-a-bug-triage-process)

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,27 +1,173 @@
...
+
+#### Spec Helpers
+
+* Use `with_modified_env` to modify an env variable for one test:
+
+```ruby
+with_modified_env INVITATION_CODE: "inv-code", INVITATIONS_ENABLED: "true" do
+post :create, user: { name: "Alice", invitation_code: "wrong-code" }
+expect(flash[:alert]).to equal "Invalid invitation code"
+end
```
Comment (jayroh) [inline CONTRIBUTING.md]: Indentation in this block

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,27 +1,173 @@
...
+</table>
+HTML
+```
+
+### Document / Photo Uploading Details
+
+* The file uploading service is called [Shubox]. The [dashboard] can be
+accessed with account info found in 1password. Contact [@jayroh] for any
+questions or issues related to Shubox.
+* The file limit is currently set to 3MB. That can be bumped up if we notice
```
Comment (jayroh) [inline CONTRIBUTING.md]: Recently changed from 3MB to 8MB.

File: README.md [hunk 3]
```diff
@@ -32,12 +55,11 @@ MiBridges::Driver.new(snap_application: snap_application).run
...
-   version, set `WEB_DRIVER` to `chrome`.
-2. If you'd like it to run a little slower, perhaps for demo purposes or to
-   look more "human", you may set `DRIVER_SPEED` to `slow` or `medium`.
-3. If you would like to display a log of some of the things happening as it
-   "drives" set `DEBUG_DRIVE` to `true`
+| Variable      | Purpose |
+| ------------- | -----------|
+| `WEB_DRIVER=chrome` | If you would like it to run in a FULL chrome browser instead of the headless version |
+| `DRIVER_SPEED=slow` or `medium` | If you'd like it to run a little slower, perhaps for demo purposes or to look more "human" |
+| `DEBUG_DRIVE=true` | If you would like to display a log of some of the things happening as it "drives" |
```
Comment (jayroh) [inline README.md]: 👍 👍 👍 👍 👍 👍

File: README.md [hunk 4]
```diff
@@ -13,15 +16,35 @@
...
+| File          | Purpose |
+| ------------- | -----------|
+| [README.md](README.md) | This root document |
+| [CONTRIBUTING.md](CONTRIBUTING.md) | How we write and test code for this application |
+| [DESIGN-README.md](DESIGN-README.md) | How we design the application |
+| [DEPLOYING.md](DEPLOYING.md) | How to deploy code to production servers |
+| [LICENSE.md](LICENSE.md) | Our software license |
+
+
+## Driving Applications
```
Comment (jessieay) [inline README.md]: it feels like this could be a part of `CONTRIBUTING` since it's something that is mostly dev-specific

File: README.md [hunk 5]
```diff
@@ -13,15 +16,35 @@
...
 
-The only requirements to run the driver code are:
+### Requirements
 
 1. A sufficiently complete `SnapApplication`
 2. Chrome and Chromedriver are installed (the latest versions). `./bin/setup`
    should get this set up for you automatically.
 
-To run the driver code you could drop into a rails console and (where
+### Driving Applications
```
Comment (jessieay) [inline README.md]: dupe header?

**Pull request merged**